src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java

```diff
@@ -195,6 +195,41 @@ public void setEt3ResponseDueDate(CaseData caseData) {
         }
     }
 
+    public void setNextListedDate(CaseData caseData) {
+        List<String> dates = new ArrayList<>();
+        String nextListedDate = "";
+
+        if (CollectionUtils.isNotEmpty(caseData.getHearingCollection())) {
+            for (HearingTypeItem hearingTypeItem : caseData.getHearingCollection()) {
+                dates = getListedDates(hearingTypeItem);
```

Comment:
is this just not overriding dates every time with the latest result from getListedDates rather than adding them in all into a list?